### DISCUSSION TITLE
Keys conversion mods break leaderboard order in osu!mania when using classic scoring

### DISCUSSION THREAD

**Discussion by respektive** [ppy/osu#17181] [Q&A]
I believe this happens, because the object count changes, which is used by the classic scoring formula.
Beatmap link: https://osu.ppy.sh/beatmapsets/1235110#mania/2569254
![](https://pek.li/xamg8r.png)

**Comment by bdach** [comment]
It will be due to that, but this leaderboard makes little sense in general if I'm being honest. Each key convert is really a different map, so this is like jamming 5 maps into one leaderboard.

In the long term I'd probably say that conversion mods should not be in the "global" default leaderboard and should only be visible via a mod leaderboard for the specific mod. Just my opinion, though.

**Reply by respektive** [reply]
Yeah, I was thinking that aswell. Personally i would only show them on the specific mod leaderboards.  

But if we decide to keep them on global, i was thinking to just use the hit objects count before the conversion for the score calculation. That should keep the order in that case. But I would like to hear more opinions on this.

**Reply by smoogipoo** [reply]
It's possible, but I'm not sure if the complexity of doing so is worth it. For scores set _during play_, it's super easy to know this value - `BeatmapConverter` can store it in `ManiaBeatmap` and `ManiaScoreProcessor` can retrieve it via `ApplyBeatmap`.

However, it would be difficult for `ManiaScoreProcessor` to determine this value _outside of play_, without having `ApplyBeatmap` called, which is where my complexity concern comes from. `BeatmapInfo` currently doesn't store this count, and the only possible source is via `APIBeatmap` right now, so we sometimes lose context of this value (when the provided beatmap comes from the local database).  
We generally should be able to recompute scores with just a `ScoreInfo` to keep things simple (see https://github.com/ppy/osu/pull/17164).

Maybe we can come back to this at a later point in time.

**Reply by respektive** [reply]
Yeah, I see the issues with it. 

Probably a good idea to wait for the survey results as well, before deciding how to deal with this.
